src/bots/discord.ts

```diff
@@ -260,76 +265,105 @@ export default class DiscordBot extends BotClient {
     });
   }
   public async start(): Promise<boolean> {
-    if (this.token) {
-      await this.bot.login(this.token);
-      this.isRunning = true;
-      // Handle being removed from a guild
-      this.bot.on('guildDelete', (guild) => {
-        const guildID = guild.id;
-        const channels = this.getBotChannels();
-
-        // Remove all channel data of that guild
-        channels.forEach((channel) => {
-          const discordChannel = this.bot.channels.cache.get(channel.id);
-          if (!discordChannel) {
-            // Can't find the channel, it probably belongs to the guild, remove data
-            return this.onRemoved(channel);
-          }
+    if (!this.enabled) {
+      this.logger.info('Autostart disabled.');
+      return false;
+    }
+    assertIsDefined(this.token, `Token is undefined`);
```

Comment:
How excactly does the bot behave when this is undefined?
Not that this will happen every time that a new contributor starts the bot for the first time.

I think I liked that it gave a warning and didn't start the bot, but didn't crash anything

Reply:
It throws an error that the main file catches. I'm fine with wrapping this in a huge try-catch but there are some nested `try-catch`s too there and it kind of looked like a mess to me